### DISCUSSION TITLE
Interest in project and offer of support

### DISCUSSION THREAD

**Discussion by funkster62** [strictdoc-project/reqif#30] [General]
I am a retired aerospace engineer with a background in MBSE that is getting back into coding after many years of mind-numbing management.  I am very interested in using NLP on poorly-written requirements so that it would be easier to integrate them into Systems Models for MBSE.  I can get specification trees in ReqIF to process but I am weak on parsing and manipulating ReqIF.  It looks as though this project would be useful to me.  I am new to Python development but would like to contribute.  Any ways I can help you with the project?  Very Best.  funkster

**Comment by stanislaw** [comment]
Hi @funkster62,

Great questions that I cannot fully answer in a single comment.

I think we can talk about two related topics:

1) How you could contribute to the `reqif` library itself.
2) How you could use the `reqif` library to do NLP on poorly-written requirements.

First, I have created this library in just a few Christmas days because I was extracting some prototype ReqIF code from the upper-level library called StrictDoc, a text-based requirements management tool that I develop as a hobby project. One of my goals is to finish all major items on the StrictDoc's ReqIF roadmap https://github.com/strictdoc-project/strictdoc/issues/520. While doing that, I would like to make the `reqif` library support full ReqIF data model and make it work with ReqIF files produced by all major requirement management tools. I didn't expect people to contribute to the library this soon, so the `reqif` does not have too many specific and actionable [Issues](https://github.com/strictdoc-project/reqif/issues) opened - this is something that I could improve on this weekend (that's when I normally have a few hours to focus on StrictDoc and now ReqIF).

Regarding the NLP: I have such a feature as a long-term goal for StrictDoc where I wanted to do at least a basic validation of how shall statements are written (one example is the EARS template, you probably know this and other papers: https://www.researchgate.net/publication/224079416_Easy_approach_to_requirements_syntax_EARS), but there are many more things to accomplish first (see https://strictdoc.readthedocs.io/en/stable/strictdoc.html#backlog).

I would assume that in your case you can either contribute the NLP extension to this (`reqif`) library directly which would be a great contribution to its feature set, or you could use the `reqif` as a low-level building block, in the same way as StrictDoc does it, and create your own library on top of it. I don't know what exactly are you planning, but I would be happy to discuss everything via Discord, see https://strictdoc.readthedocs.io/en/latest/strictdoc.html#contact-the-developers. You are welcome to write to the #general channel or just message me there directly.

**Reply by funkster62** [reply]
Stanislaw,  I crammed python, Git, PyCharm.... the whole toolchain over Christmas break because after a few months of working part time, I saw a need for some automation in tool to tool data transfer.  I work with people who manage requirements in DOORS and people who do MBSE in Cameo Systems Modeler.  The bridge between the two tools is ReqIF but the semantic precision of DOORS data is not typically to the level that can be used in a modeling tool.  

As a senior systems engineer I am am way down the learning curve on Python and app development workflow.  Most of my work over the past decade was working with large scale projects doing MBSE using SysML tools.  I worked on system models, not code but now that I do not have management responsibility, I can have fun again! :) I will spend the next couple of days reviewing your project to try to understand the architecture and coding.  If you would like, I can work on comments or documentation at some point.  I just got into Git over the past month so I need to get up on Git workflow before I trust myself to add anything to the code level.  What I could do is come up with notes on the code and feed them back to you for inclusion or correction.  Very Best! funkster

**Comment by stanislaw** [comment]
Hi @funkster62, I hope you are doing well. It's been a while since we last talked. I am going through the open discussion threads and closing those that are not active anymore. I am going to close this one, but I would be happy to hear from you if you are still working with the ReqIF/requirements topic.
